week-1/fix/median.js

```diff
@@ -3,9 +3,18 @@
 // If you're in the week-1 directory, you can run npm test -- fix to run the tests in the fix directory
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+  const sortedList = list.slice().sort((a, b) => a - b);
+
+  const middleIndex = Math.floor(sortedList.length / 2);
+
+  if (sortedList.length % 2 === 0) {
+    const middleValues = sortedList.slice(middleIndex - 1, middleIndex + 1);
+    return (middleValues[0] + middleValues[1]) / 2;
+  } else {
+    return sortedList[middleIndex];
+  }
+  // const median = list.splice(middleIndex, 1)[0];
```

Comment:
Great job, you can remove these lines since it is not useful?
Is not good practice to push commented code.